### ISSUE TITLE
project directory can't have same name as a dependency used by project

### ISSUE THREAD

**Issue by zachvalenta** [python-poetry/poetry#3438]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS Mojave 10.14.6 18G103 x86_64
- **Poetry version**: Poetry version 1.1.4
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/zachvalenta/1f7051519003d41a568fbd3ea73f8a99

## Issue

I noticed a Poetry project with the same name as a dependency used by the project seems to cause a conflict when downloading that dependency.

For example, I wanted to try out the BDD library [behave](https://github.com/behave/behave) and so created a new directory also called `behave`.
```sh
$ mkdir behave
$ cd behave
$ poetry init -n
$ poetry add -vvv behave

Using version ^1.2.6 for behave

Updating dependencies
Resolving dependencies... (0.0s)

  AssertionError

  at ~/.local/pipx/venvs/poetry/lib/python3.6/site-packages/poetry/mixology/incompatibility.py:111 in __str__
      107│         )
      108│
      109│     def __str__(self):
      110│         if isinstance(self._cause, DependencyCause):
    → 111│             assert len(self._terms) == 2
      112│
      113│             depender = self._terms[0]
      114│             dependee = self._terms[1]
      115│             assert depender.is_positive()
```

But when I rename the directory (and Poetry project) to `test-behave` this problem goes away. I tried the same thing with Django and found the same behavior. I don't know if this is a bug exactly; naming your project directory so uncreatively is an anti-pattern anyway and I was doing so only for a quick proof-of-concept. Still, figure other lazy namers might run into this issue :)

**Comment by sinoroc** [comment]
On some levels, your own project and its dependencies (or any other project installed in the same environment) are treated the same. So basically there are 2 projects with the same name that are supposed to be installed in 1 environment, this can not possibly work.

_poetry_ could maybe deliver more informative error message, maybe, but really in my opinion that would be adding more code complexity to _poetry_ for little gain.

**Comment by zachvalenta** [comment]
Yeah I was definitely thinking of the 'solution' to this as a little notice in the docs vs. code change. Both plausible I suppose but docs feel more appropriate.

**Comment by cocoonkid** [comment]
Oh thanks. I had the same problem. A better error message would be great yes :-0

**Comment by xeor** [comment]
Just linking this issue to https://github.com/python-poetry/poetry/issues/1051#issuecomment-603245885 as many people want this, even tho it doesnt reflect that on this issue only.

There are going to be situations where you want the same name, example when you are not using poetry not primary for packaging, but for keeping a version, example if I got a `my-platform/ansible` folder, and inside the `ansible` folder I want poetry to keep track of the package called `ansible`

**Comment by ndevenish** [comment]
Just had a confused 20 minutes tracking down to this issue. I wanted to try out using a package, so made a folder named the same thing.

**Comment by takeda** [comment]
Just got this error. I see this open for 2 years so far. At least if there's no intention of fixing it, adding a an error message that states what's going on would help greatly.

**Comment by xhiroga** [comment]
I guess this PR https://github.com/python-poetry/poetry/pull/3841 resolve this issue.

**Comment by radoering** [comment]
Duplicate of #236, resolved by #3841
